processing/src/main/java/io/druid/query/dimension/RegexFilteredDimensionSpec.java

```diff
@@ -95,7 +95,7 @@ public IndexedInts getRow()
       @Override
       public int getValueCardinality()
       {
-        return bitSetOfIds.cardinality();
+        return selector.getValueCardinality();
```

Comment:
this breaks contract of getValueCardinality(), this should return cardinality "after" filtering.